### PR TITLE
Clean up obsolete Bash coding style in bootstrap.sh

### PR DESCRIPTION
**Pull request by janosgyerik**

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
 		--exclude "LICENSE-MIT.txt" \
-		-avh --no-perms . ~;
-	source ~/.bash_profile;
+		-avh --no-perms . ~
+	source ~/.bash_profile
 }
 
-if [ "$1" == "--force" -o "$1" == "-f" ]; then
-	doIt;
+if [[ $1 == --force || $1 == -f ]]; then
```
Comment (mathiasbynens) [inline bootstrap.sh]: > Replace undocumented and obsolescent elements

Can you elaborate on this? What makes you say these are undocumented?

Comment (janosgyerik) [inline bootstrap.sh]: Sure! If you take a look at `man [`, the documentation of the `[` command, it explains the `=` operator, but the `==` operator is nowhere to be found. It's a mystery it works at all.

The `==` operator is explained for the more modern `[[` command in `help [[`.

And the `-o` operator is not recommended.

See also the comments on [this question](http://stackoverflow.com/q/38682618/641955) and [its answer](http://stackoverflow.com/a/38684324/641955) on Stack Overflow.

**Comment by Tatsh** [comment]
If you are going to fix this line, why not make it right by `shellcheck -s bash`?

```bash
if [[ "$1" = '--force' ]] || [[ "$1" = '-f' ]]; then
```

Reading when `-o` or `||` is between is `[[ ]]` is more difficult. And you should always quote variables except when you want expansion.

`=`/`==`/ is [covered in the Bash Wiki](http://mywiki.wooledge.org/BashFAQ/031). This script targets Bash, and `[[` (therefore `==`) is understood by Bash. Only if you use `[` must you use single `=`. In this project I do not think we are concerned about really old shells or alternatives like Dash.

**Comment by janosgyerik** [comment]
@Tatsh For the record, `shellcheck -s bash bootstrap.sh` reported nothing to me. But I took your advices and changed the line as you suggested, thanks!

**Comment by Tatsh** [comment]
Hmm that is true. `shellcheck` tells me to get rid of `-o` and quote the variables to prevent globbing. But it does not care about `||` inside of `[[ ]]`. Also checked bashate and it too does not care. So I guess I just read this as 'best practises' somewhere and I've stuck with it ever since. I am sure my older scripts have horrible unreadable lines with lots of `-o` or `||` inside single `[[ ]]` tests.

**Comment by janosgyerik** [comment]
@Tatsh Looks like you like this stuff, so I'll share a bit more details with you ;-)

I moved the `||` inside as you suggested because otherwise the highlighting in Vim was not great. That was already a reason enough for me.

I disagreed with "you should always quote variables except when you want expansion". For one thing, I think you should always quote variables that may contain spaces, typically filesystem paths. Another thing, in your example you quoted `-f` and `--force` which are not variables but literal strings. Lastly, the `$1` at the left side don't need to be quoted at all. But I took your suggestion anyway, because consistently quoting everything invites less questions and doubts, which is a good thing. And although it's unnecessary to quote literal strings like `-f` and `--force`, again Vim highlighting works better if I quote them.

The commits and their messages that I pushed after your suggestion reflect my thought process and how I came to change the line exactly as you suggested. Thanks for the discussion!

**Comment by janosgyerik** [comment]
@mathiasbynens any other questions or doubts?

**Comment by mathiasbynens** [comment]
I appreciate the patch, but I like my current Bash coding style.

**Pull request closed without merging**